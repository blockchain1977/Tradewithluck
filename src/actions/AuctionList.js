import petData from '../assets/pets.json';
import Adoption from '../blockchain/Adoption'

export function getAuctions() {
  const url = 'http://ec2-54-92-55-120.ap-northeast-1.compute.amazonaws.com:3000/';

  Adoption.getallAdoptedStatus().then(status => dispatch =>
    Promise.resolve(
        dispatch({
          type: 'AUCTIONS_UPDATE',
          payload: petData.map(pet => ({
            key: pet.id,
            name: pet.name,
            image: url + pet.picture,
            age: pet.age,
            breed: pet.breed,
            location: pet.location,
            adopted: status[parseInt(pet.id, 10)]
          }))
        })
      ));
}

export function toggleFAB() {
  return dispatch =>
    // new Promise(resolve =>
    // resolve(
    dispatch({
      type: 'TOGGLE_FAB',
      payload: ''
    });
  // )
  // ).catch(e => console.log(e));
}
