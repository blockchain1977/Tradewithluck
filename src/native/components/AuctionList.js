import React from 'react';
import { FlatList, TouchableOpacity } from 'react-native';
import { Container, Content, Text, Card, CardItem, View } from 'native-base';

import PropTypes from 'prop-types';

const AuctionList = ({
  auctions,
}) => (
  <Container>
    <Content>
      <FlatList
        numColumns={1}
        data={auctions}
        renderItem={({ item }) => (
          <Card>
            <CardItem cardBody>
              <TouchableOpacity>
                <View style={{ flex: 1, height: 30 }} />
                <Text style={{ fontWeight: '800' }}>{item.key}</Text>
              </TouchableOpacity>
            </CardItem>
          </Card>
        )}
      />
    </Content>
  </Container>
);


AuctionList.propTypes = {
  auctions: PropTypes.arrayOf(PropTypes.shape()).isRequired,
};

export default AuctionList;
