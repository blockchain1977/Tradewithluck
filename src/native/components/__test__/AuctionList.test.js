import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// imported as a connected component!
import AuctionList from '../AuctionList';

configure({ adapter: new Adapter() });

const setup = state => {
  const actions = { toggleFAB: jest.fn(), adoptPet: jest.fn() };
  const component = shallow(
    <AuctionList auctions={state.auctions} fabstatus={state.fabstatus} {...actions} />
  );

  return {
    component,
    actions,
    fab: component.find('[position="bottomRight"]')
  };
};

describe('Testing AuctionList', () => {
  let initialState;

  beforeEach(() => {
    initialState = {
      auctions: [
        {
          key: 0,
          name: 'Kyle',
          image: 'http://52.199.0.93:3000/images/scottish-terrier.jpeg',
          age: '5',
          breed: 'Test breed',
          location: 'Auckland',
          adopted: false
        }
      ],
      fabstatus: false
    };
  });

  it('check rendered', () => {
    const { component } = setup(initialState);
    expect(component.dive()).toMatchSnapshot();
  });

  it('check rendered 2', () => {
    const { component } = setup({ auctions: initialState.auctions, fabstatus: true });
    expect(component.dive()).toMatchSnapshot();
  });
});
