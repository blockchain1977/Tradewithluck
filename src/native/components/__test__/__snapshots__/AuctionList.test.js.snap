// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AuctionList check rendered 1`] = `
<Container
  style={
    Object {
      "flex": 1,
      "height": 1334,
    }
  }
>
  <Styled(Content)>
    <FlatList
      data={
        Array [
          Object {
            "adopted": false,
            "age": "5",
            "breed": "Test breed",
            "image": "http://52.199.0.93:3000/images/scottish-terrier.jpeg",
            "key": 0,
            "location": "Auckland",
            "name": "Kyle",
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      renderItem={[Function]}
      scrollEventThrottle={50}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </Styled(Content)>
  <Styled(ViewNB)>
    <Styled(Fab)
      active={false}
      containerStyle={Object {}}
      direction="up"
      onPress={[Function]}
      position="bottomRight"
      style={
        Object {
          "backgroundColor": "#5067FF",
        }
      }
    >
      <Styled(Icon)
        name="share"
      />
      <Styled(Button)
        style={
          Object {
            "backgroundColor": "#34A34F",
          }
        }
      >
        <Styled(Icon)
          name="logo-whatsapp"
        />
      </Styled(Button)>
      <Styled(Button)
        style={
          Object {
            "backgroundColor": "#3B5998",
          }
        }
      >
        <Styled(Icon)
          name="logo-facebook"
        />
      </Styled(Button)>
      <Styled(Button)
        disabled={true}
        style={
          Object {
            "backgroundColor": "#DD5144",
          }
        }
      >
        <Styled(Icon)
          name="mail"
        />
      </Styled(Button)>
    </Styled(Fab)>
  </Styled(ViewNB)>
</Container>
`;

exports[`Testing AuctionList check rendered 2 1`] = `
<Container
  style={
    Object {
      "flex": 1,
      "height": 1334,
    }
  }
>
  <Styled(Content)>
    <FlatList
      data={
        Array [
          Object {
            "adopted": false,
            "age": "5",
            "breed": "Test breed",
            "image": "http://52.199.0.93:3000/images/scottish-terrier.jpeg",
            "key": 0,
            "location": "Auckland",
            "name": "Kyle",
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      renderItem={[Function]}
      scrollEventThrottle={50}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </Styled(Content)>
  <Styled(ViewNB)>
    <Styled(Fab)
      active={true}
      containerStyle={Object {}}
      direction="up"
      onPress={[Function]}
      position="bottomRight"
      style={
        Object {
          "backgroundColor": "#5067FF",
        }
      }
    >
      <Styled(Icon)
        name="share"
      />
      <Styled(Button)
        style={
          Object {
            "backgroundColor": "#34A34F",
          }
        }
      >
        <Styled(Icon)
          name="logo-whatsapp"
        />
      </Styled(Button)>
      <Styled(Button)
        style={
          Object {
            "backgroundColor": "#3B5998",
          }
        }
      >
        <Styled(Icon)
          name="logo-facebook"
        />
      </Styled(Button)>
      <Styled(Button)
        disabled={true}
        style={
          Object {
            "backgroundColor": "#DD5144",
          }
        }
      >
        <Styled(Icon)
          name="mail"
        />
      </Styled(Button)>
    </Styled(Fab)>
  </Styled(ViewNB)>
</Container>
`;
